src/fmt.rs

```diff
@@ -0,0 +1,210 @@
+use opentelemetry::trace::TraceContextExt;
+use serde::{
+    Deserialize, Deserializer as _, Serialize, Serializer as _,
+    de::{Error, MapAccess, Visitor as DeVisitor},
+    ser::{SerializeMap, SerializeSeq},
+};
+use serde_json::{Deserializer, Serializer, Value};
+use std::{fmt, io, marker::PhantomData, ops::Deref, str};
+use tracing::{Event, Span, Subscriber};
+use tracing_opentelemetry::OpenTelemetrySpanExt;
+use tracing_serde::{AsSerde, SerdeMapVisitor};
+use tracing_subscriber::{
+    fmt::{
+        FmtContext, FormatEvent, FormatFields, FormattedFields,
+        format::Writer,
+        time::{FormatTime, SystemTime},
+    },
+    registry::{LookupSpan, SpanRef},
+};
+
+pub struct JsonFormat;
+
+impl<S, N> FormatEvent<S, N> for JsonFormat
+where
+    S: Subscriber + for<'lookup> LookupSpan<'lookup>,
+    N: for<'writer> FormatFields<'writer> + 'static,
+{
+    fn format_event(
+        &self,
+        ctx: &FmtContext<'_, S, N>,
+        mut writer: Writer<'_>,
+        event: &Event<'_>,
+    ) -> fmt::Result {
+        let mut timestamp = String::new();
+        SystemTime.format_time(&mut Writer::new(&mut timestamp))?;
+
+        let meta = event.metadata();
+
+        let mut visit = || {
+            let mut serializer = Serializer::new(IoWriter(&mut writer));
+            let mut serializer = serializer.serialize_map(None)?;
+
+            serializer.serialize_entry("timestamp", &timestamp)?;
+            serializer.serialize_entry("level", &meta.level().as_serde())?;
+
+            // add all event fields to the json object
+            let mut visitor = SerdeMapVisitor::new(serializer);
+            event.record(&mut visitor);
+            serializer = visitor.take_serializer()?;
```

Comment:
I just realised that flattening the log entry may result in having duplicate json keys as well.

And this is also exactly the same for `tracing_subscriber`.

Reply:
Apparently, there is also a way to duplicate "message": https://github.com/tokio-rs/tracing/issues/3195